### PR TITLE
soc: Support for qn9090 and dk6_qn9090

### PR DESCRIPTION
**Pull request by HesselM**
Adds support for the qn9090 soc of NXP and dev board dk6_qn9090
hal_nxp PR: https://github.com/zephyrproject-rtos/hal_nxp/pull/294

### CODE REVIEW

File: boards/nxp/dk6_qn9090/Kconfig.defconfig [hunk 1]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2023 Sendrato
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if BOARD_DK6_QN9090
+
+# no configurations
```
Comment (nordicjm) [inline boards/nxp/dk6_qn9090/Kconfig.defconfig:9]: remove empty files

File: boards/nxp/dk6_qn9090/Kconfig.dk6_qn9090 [hunk 2]
```diff
@@ -0,0 +1,11 @@
+#
+# Copyright (c) 2023 Sendrato
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config BOARD_DK6_QN9090
+	bool
```
Comment (nordicjm) [inline boards/nxp/dk6_qn9090/Kconfig.dk6_qn9090:8]: adapt to hwmv2

File: boards/nxp/dk6_qn9090/board.yml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+board:
+  name: dk6_qn9090
```
Comment (nordicjm) [inline boards/nxp/dk6_qn9090/board.yml:2]: is the soc name really needed in the board?

File: boards/nxp/dk6_qn9090/doc/QN9090-DK006-BD.jpeg
Comment (nordicjm) [inline boards/nxp/dk6_qn9090/doc/QN9090-DK006-BD.jpeg:1]: Convert to webp to reduce filesize

File: drivers/bluetooth/hci/CMakeLists.txt [hunk 4]
```diff
@@ -33,3 +33,8 @@ zephyr_library_sources_ifdef(CONFIG_BT_SILABS_HCI  slz_hci.c)
 zephyr_library_sources_ifdef(CONFIG_BT_PSOC6_BLESS hci_psoc6_bless.c)
 zephyr_library_sources_ifdef(CONFIG_SOC_NRF5340_CPUAPP nrf53_support.c)
 zephyr_library_sources_ifdef(CONFIG_BT_AMBIQ_HCI   hci_ambiq.c apollox_blue.c)
+zephyr_library_sources_ifdef(CONFIG_BT_NXP_IPM     ipm_nxp.c)
+
+if ((CONFIG_CODE_DATA_RELOCATION) AND (CONFIG_SOC_QN9090))
```
Comment (nordicjm) [inline drivers/bluetooth/hci/CMakeLists.txt:38]:
```suggestion
if((CONFIG_CODE_DATA_RELOCATION) AND (CONFIG_SOC_QN9090))
```

File: drivers/bluetooth/hci/Kconfig [hunk 5]
```diff
@@ -234,3 +241,21 @@ config BT_SILABS_HCI_BUFFER_MEMORY
...
+	int "BT NXP IPM init priority"
+	default 75
+
+if SOC_FAMILY_K32
+
+config BT_NXP_FIXED_MAC_ADDR
+	hex "Fixed BT-MAC address of k32 soc"
+	default 0
+	help
+		6-byte hexadecimal address, eg: 0xffaa01610401.
```
Comment (nordicjm) [inline drivers/bluetooth/hci/Kconfig:257]: help text indent is tab followed by 2 spaces

File: drivers/bluetooth/hci/ipm_nxp.c [hunk 6]
```diff
@@ -0,0 +1,332 @@
...
+#define HCI_ACL 0x02
+#define HCI_SCO 0x03
+#define HCI_EVT 0x04
+
+#define ACI_WRITE_SET_TX_POWER_LEVEL BT_OP(BT_OGF_VS, 0xFC0F)
+
+/* TODO: get proper size from bt_controller_task_config.h */
+#define BLE_CONTROLLER_STACK_SIZE (4096)
+
+/* TODO: map this to config */
```
Comment (nordicjm) [inline drivers/bluetooth/hci/ipm_nxp.c:32]: should be done?

File: soc/nxp/k32/Kconfig [hunk 7]
```diff
@@ -0,0 +1,24 @@
+# NXP K32 family
+
+#
+# Copyright (c) 2023 Sendrato
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+config SOC_FAMILY_K32
+	bool
```
Comment (nordicjm) [inline soc/nxp/k32/Kconfig:10]:
```suggestion
```

**Review by gmarull — CHANGES_REQUESTED** [review]
please, no single commits, check our contrib guidelines

File: soc/nxp/k32/k32/Kconfig.defconfig.qn9090 [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+#
+# Copyright (c) 2023 Sendrato
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+
+if SOC_QN9090
+
+config SOC
+	default "qn9090"
```
Comment (nordicjm) [inline soc/nxp/k32/k32/Kconfig.defconfig.qn9090:12]: move to Kconfig.soc

File: subsys/bluetooth/Kconfig [hunk 9]
```diff
@@ -127,8 +127,8 @@ config BT_CONN_TX
 if BT_CONN
 config BT_HCI_ACL_FLOW_CONTROL
 	bool "Controller to Host ACL flow control support"
-	# Enable if building a Host-only build
-	default y if !BT_CTLR && !BT_STM32_IPM && !BT_ESP32  && !BT_STM32WBA
+	# Enable if building a Host-only builds
```
Comment (alwa-nordic) [inline subsys/bluetooth/Kconfig:130]: typo

**Review by dleach02 — CHANGES_REQUESTED** [review]
NXP is not prepared at this moment to support qn9090 in our HAL. There are also other issues as the HAL PR includes a number of the binary blobs needed to enable the BLE controller which are not allowed directly to Zephyr upstream.

**Comment by dleach02** [comment]
@HesselM, Thank you for the contribution. 

I'm putting a block on this PR as there are larger issues to resolved from a NXP perspective. When the community desires to add NXP SOC support to our HAL, we have to consider whether we are prepared to provide long term support for the enablement. Also, there are structural problems with the HAL side PR as it includes a number of binary libraries that are not allowed by the Zephyr Project in the upstream.

My recommendation is for you to host your own downstream for this enablement until we can consider and/or resolve the structural and support issues.

**Pull request closed without merging**